store/node.go

```diff
@@ -0,0 +1,265 @@
+package store
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+func (iter nodeIterator) nodePtr() *nodePtr {
+	if !iter.Valid() {
+		return nil
+	}
+	res := nodePtr{
+		tree:  iter.tree,
+		level: iter.level,
+		key:   iter.Key()[7:],
+	}
+	return &res
+}
+
+func (nodePtr *nodePtr) isLeaf() bool {
+	return nodePtr.level == 0
+}
+
+func (nodePtr *nodePtr) children() (res children) {
+	bz := nodePtr.tree.store.Get(nodePtr.tree.nodeKey(nodePtr.level, nodePtr.key))
+	if bz != nil {
+		json.Unmarshal(bz, &res)
+	}
+	return
+}
+
+func (nodePtr *nodePtr) set(children children) {
+	bz, err := json.Marshal(children)
+	if err != nil {
+		panic(err)
+	}
+	nodePtr.tree.store.Set(nodePtr.tree.nodeKey(nodePtr.level, nodePtr.key), bz)
+}
+
+func (nodePtr *nodePtr) setLeaf(acc uint64) {
+	if !nodePtr.isLeaf() {
+		panic("setLeaf should not be called on branch nodePtr")
+	}
+	bz, err := json.Marshal(acc)
+	if err != nil {
+		panic(err)
+	}
+	nodePtr.tree.store.Set(nodePtr.tree.leafKey(nodePtr.key), bz)
+}
+
+// delete removes the corresponding node from the underlying data store,
+func (nodePtr *nodePtr) delete() {
+	nodePtr.tree.store.Delete(nodePtr.tree.nodeKey(nodePtr.level, nodePtr.key))
+}
+
+func (nodePtr *nodePtr) leftSibling() *nodePtr {
+	return nodePtr.tree.nodeReverseIterator(nodePtr.level, nil, nodePtr.key).nodePtr()
+}
+
+func (nodePtr *nodePtr) rightSibling() *nodePtr {
+	iter := nodePtr.tree.nodeIterator(nodePtr.level, nodePtr.key, nil)
+	if !iter.Valid() {
+		return nil
+	}
+	if nodePtr.exists() {
+		// exclude nodePtr itself
+		iter.Next()
+	}
+	return iter.nodePtr()
+}
```

Comment:
Why is the left sibling code different than the right sibling code, other than the nodeIteratos vs nodeReverseIterator line?